sysquant/optimisation/generic_optimiser.py

```diff
@@ -50,23 +46,17 @@ def returns_processor(self) -> returnsPreProcessor:
     def weighting_params(self) -> dict:
         return self._weighting_params
 
-    @property
-    def optimiser_over_time(self) -> optimiseWeightsOverTime:
-        return self._optimiser_over_time
-
     def weights(self) -> pd.DataFrame:
-
         raw_weights = self.raw_weights()
         weights = self.weights_post_processing(raw_weights)
 
         ## apply cost weight
         return weights
 
     def raw_weights(self) -> pd.DataFrame:
-        optimiser = self.optimiser_over_time
-        weights = optimiser.weights()
+        optimiser = optimiseWeightsOverTime(self.net_returns, log=self.log, **self.weighting_params)
```

Comment:
This doesn't make sense to me. The optimiser is set when the object is created, why create it anew? I use this particular pattern deliberately as it makes it easier to interrogate the optimiser objects and understand what's going on.

Reply:
It is a valid point, but if we hold the reference to the `optimiser`, it won't be garbage-collected. We are talking tens (or hundreds) of GBs here, as we initialize the `optimiser` for each instrument and it recalculates and stores the whole correlation matrix for all the instruments.

Looking at https://github.com/robcarver17/pysystemtrade/blob/master/sysquant/optimisation/optimise_over_time.py there isn't much to inspect tbh, except for `fit_dates` maybe. How about I extract it to a separate method, so a new instance can be instpected?

Reply:
OK that makes sense, but why don't you delete these lines from the __init__ for generic_optimiser as they are now redundant:

```
        optimiser_over_time = optimiseWeightsOverTime(net_returns, log = log,
                                          **weighting_params)
        self._optimiser_over_time = optimiser_over_time

```

Reply:
... and the method 

```
    @property
    def optimiser_over_time(self) -> optimiseWeightsOverTime:
        return self._optimiser_over_time

```

Reply:
I'm a bit lost here… I believe I did just that… https://github.com/robcarver17/pysystemtrade/pull/373/files#diff-0fcad28880fb200f3124449d3cfe57fab9c3c486fc6ee6f9c2d88a8e402bb3faL17 and https://github.com/robcarver17/pysystemtrade/pull/373/files#diff-0fcad28880fb200f3124449d3cfe57fab9c3c486fc6ee6f9c2d88a8e402bb3faL53

Reply:
No it's me that's lost! I missed that. Thanks for the pull.